src/model.rs

```diff
@@ -143,9 +138,9 @@ impl Builder {
             i += 1;
 
             if i % self.checkpoint_interval == 0 {
-                println!("");
-                println!(" ================== Iteration {} ==================", i);
-                println!("");
+                info!("");
+                info!(" ================== Iteration {} ==================", i);
+                info!("");
```

Comment:
I think the header should be printed as is anyway at the info level (same for the thread one) because it's easier to read logs with it at the trace level. Also, at the info level this still provides
useful high-level information about the way threads/iterations are scheduled, without the overhead of trace messages.

As for using spans for iterations and threads, I'm all for it, but I think it should be done separately since (1) it requires some changes to pass spans around, (2) I don't know how much time I can spend on this PR, and (3) there's barely any logging in master now.

Reply:
> As for using spans for iterations and threads, I'm all for it, but I think it should be done separately since (1) it requires some changes to pass spans around, (2) I don't know how much time I can spend on this PR, and (3) there's barely any logging in master now.

I totally agree, this was intended as a suggestion for a future branch! Let's move forwards with this as-is and add spans afterwards.